src/pages/iv-items/forms/SalesForm.js

```diff
@@ -0,0 +1,250 @@
+import { useState, useContext } from 'react'
+import { Form, useFormik } from 'formik'
+import { Grid, FormControlLabel, Checkbox } from '@mui/material'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useEffect } from 'react'
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { useForm } from 'src/hooks/form'
+import { useInvalidate } from 'src/hooks/resource'
+import { ControlContext } from 'src/providers/ControlContext'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { DataSets } from 'src/resources/DataSets'
+
+const SalesForm = ({ labels, maxAccess, store, record, cId }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.Sales.qry
+  })
+
+  const { recordId: itemId } = store
+
+  const validationSchema = yup.object({})
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      itemId,
+      currencyId: cId,
+      plId: '',
+      priceType: '',
+      valueType: '',
+      value: '',
+      priceWithVat: '',
+      minPrice: '',
+
+      ...record
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      currencyId: yup.string().required(),
+      plId: yup.string().required(),
+      priceType: yup.string().required(),
+      value: yup.string().required(),
+      valueType: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const plId = formik.values.plId
+      const currencyId = formik.values.currencyId
+
+      const response = await postRequest({
+        extension: SaleRepository.Sales.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!plId && !currencyId) {
+        toast.success(platformLabels.Added)
+      } else toast.success(platformLabels.Edited)
+
+      formik.setValues(obj)
+
+      invalidate()
+    }
+  })
+  const editMode = !!record
+
+  useEffect(() => {
```

Comment:
refactor useEffect as I mention before in SaleList (no need to create a function and recall it)